### PR TITLE
Fixed profile and auth UI

### LINKED ISSUE #175: Fix profile pages UI
The profile pages still have a very clunky UI that is basically a holdover from mitxpro. These should be updated to match the styling of the auth pages, and the "Continue" button should be changed to "Submit"

ss 2021-09-10 at 07 50 46
ss 2021-09-10 at 07 51 01
ss 2021-09-10 at 07 51 31

### LINKED ISSUE #178: Design regression on Authentication, Registration Profile forms
### Steps to Reproduce
- Visit any of these links ([Sign In](https://rc.mitxonline.mit.edu/signin/), [Register](https://rc.mitxonline.mit.edu/create-account/), [Forgot Password](https://rc.mitxonline.mit.edu/signin/forgot-password/)) in Mobile view
- You can check that the form width matches the window and it gives non-card look and feel
- You can see the screenshots below

**Note:**
- I suspect that this issue will also be on settings, profile pages as well, so you might want to take a look at them too.
- I couldn't find a related Invision design so I didn't add a link to that.

### Expected Behavior
- There should be some horizontal margin to the forms container so that it gives a card like look & feel similar to the desktop

### Actual Behavior
- The container matches the width of the window and it loses the card like feel

### Screenshot or Screencast
Screenshot 2021-09-10 at 7 42 35 PM
Screenshot 2021-09-10 at 7 42 55 PM
Screenshot 2021-09-10 at 7 44 15 PM

### PR DESCRIPTION
**Pull request by gsidebo**
#### Pre-Flight checklist

- [x] Screenshots and design review for any changes that affect layout or styling
  - [x] Desktop screenshots
  - [x] Mobile width screenshots
- [x] Testing
  - [x] Changes have been manually tested

#### What are the relevant tickets?
Fixes #175 
Fixes #178 

#### What's this PR do?
- Gets rid of the profile view page, and changes the profile link to go directly to the profile edit page
- Fixes the styling for the profile/account settings pages and all of the auth pages so that the spacing is consistent and the cards still show up as expected in mobile screen widths
- Makes some common sense fixes (e.g.: moving the title inside of the card and changing the buttons on profile/account pages to match the styling of the auth pages, getting rid of unnecessary HTML elements, incorrect grammar in error messages)

#### How should this be manually tested?
This PR touches every auth page and the profile/account settings pages, so a thorough functional test would include looking at all of these pages and making sure they look good in for all screen widths:

- Sign in
- Create account
- Create account "thank you" (where you land after you submit an email to the Create Account form)
- Forgot password
- Register details (where you land when you click the email verification link)
- Edit profile
- Edit account settings

#### Any background context you want to provide?
- The profile/account settings pages were not included in the mitxonline designs, so we kept a lot of styling from mitxpro for those pages. To me it made sense to match the styling of those pages/forms to the auth pages, which _do_ have designs for mitxonline, so I made a lot of judgment calls to work toward that goal

#### Screenshots (if appropriate)

**AUTH PAGES (mobile)**
ss 2021-09-15 at 15 51 55
ss 2021-09-15 at 15 51 48
ss 2021-09-15 at 15 53 58
ss 2021-09-15 at 15 53 19

**PROFILE/ACCOUNT PAGES**
ss 2021-09-15 at 15 39 43
ss 2021-09-15 at 15 39 58
ss 2021-09-15 at 15 40 31
ss 2021-09-15 at 15 40 39

### CODE REVIEW

**Comment by pdpinch** [comment]
Does this fix https://github.com/mitodl/mitxonline/issues/178 as well?

**Comment by gsidebo** [comment]
@pdpinch It does. Thanks for the reminder. I forgot to put that in the PR description

**Comment by pdpinch** [comment]
We got designs for these pages now. If it's not too late, can we incorporate them? 

https://invis.io/ZF11PHMQJ6CK#/458318719_MITx_-_Profile_-_Edit
https://invis.io/KH11PMX9ZRQE#/458318720_MITx_-_Account
https://invis.io/EF11PMX5X3TY#/458325008_Username_States

**Comment by pdpinch** [comment]
Do you know how removing the static profile page interacts with open edX?

I'm not sure, but I think there are two places where open edX might link to a profile:
- user menu, where linking to a profile edit form makes sense
- forum post, which probably isn't working at all

now that I write this down, I think it's out of scope for this PR, but we should take a look at open an necessary follow-up issues.

**Comment by gsidebo** [comment]
@pdpinch I actually wasn't aware that edX might have links to a profile page. I can add an issue to explore that. Re: the designs, I think the only impact on this PR is a label on one of those fields + a header text change. I'll update this. I'm also pretty certain that you don't want to mess around with a user-uploaded profile image in this PR, so we can discuss that for a separate issue

**Comment by gsidebo** [comment]
@pdpinch Screenshots updated

**Comment by umarmughal824** [comment]
@pdpinch could you please review the updated screenshots?

File: static/scss/product-details.scss [hunk 1]
```diff
@@ -11,6 +11,13 @@
     margin: 0 auto;
   }
 
+  .btn-primary.highlight {
```
Comment (umarmughal824) [inline static/scss/product-details.scss:14]: should this not come in the `common.scss` file in `.btn-primary` class?

**Review by umarmughal824 — CHANGES_REQUESTED** [review]
i have some minor concerns

File: static/scss/product-details.scss (hunk 1 above)
Comment (gsidebo) [reply static/scss/product-details.scss:14]: It's true that only the product detail page has this button with the `highlight` class, but the spacing of this button is specific to the product detail page, and we may have other pages in the future where we want a `highlight` button but we don't want the same spacing.

File: static/js/components/forms/ChangeEmailForm.js [hunk 2]
```diff
@@ -53,23 +55,27 @@ const ChangeEmailForm = ({ onSubmit, user }: Props) => (
             <ErrorMessage name="email" component={FormError} />
           </div>
           <div className="form-group">
-            <label htmlFor="confirmPassword">Confirm Password</label>
+            <label htmlFor="confirmPassword" className="row">
```
[the remark this answers is not in the document]
Comment (gsidebo) [reply static/js/components/forms/ChangeEmailForm.js:58]: That `row` is what makes the label text and the `subtitle` appear on the same line on wide screen widths. If we remove it, the label text and the subtitle always have their own line. We do the same thing in a couple places in `static/js/components/forms/ProfileFormFields.js`

**Review by umarmughal824 — CHANGES_REQUESTED** [review]
Everything looks good except the Password Reset page

@gsidebo  I have thoroughly tested your PR, I could see design issues in the Reset Password page. I am not sure if it's a part of the ticket or not.

**Note:** Please let me know if it's not a part of the ticket or are you willing to fix it in another PR so that you could merge and we could move forward.

Screenshot 2021-09-20 at 17 42 27
Screenshot 2021-09-20 at 17 42 34
Screenshot 2021-09-20 at 17 42 41

**Comment by gsidebo** [comment]
@umarmughal824 great catch! I'll take a look

**Review by umarmughal824 — APPROVED** [review]
Looks good 👍

**Pull request merged**